api/server.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
+	ds "github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log"
+	tserv "github.com/textileio/go-textile-core/threadservice"
+	pb "github.com/textileio/go-textile-threads/api/pb"
+	es "github.com/textileio/go-textile-threads/eventstore"
+	"github.com/textileio/go-textile-threads/util"
+	logger "github.com/whyrusleeping/go-logging"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.Logger("api")
+)
+
+type Server struct {
+	rpc     *grpc.Server
+	proxy   *http.Server
+	service *service
+
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+type Options struct {
+	Addr      string // defaults to 0.0.0.0:9090
+	ProxyAddr string // defaults to 0.0.0.0:9091
+	Debug     bool
+}
+
+func NewServer(ctx context.Context, ds ds.Datastore, ts tserv.Threadservice, opts Options) (*Server, error) {
+	var err error
+	if opts.Debug {
+		err = util.SetLogLevels(map[string]logger.Level{
+			"api": logger.DEBUG,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	s := &Server{
+		rpc: grpc.NewServer(),
+		service: &service{
+			datastore: ds,
+			threads:   ts,
+			stores:    make(map[string]*es.Store), // @todo how to reload stores on start?
```

Comment:
Good question. Maybe inventing some `StoreManager` with a simple badger to stor store IDs and they repo paths.